### PR TITLE
Shave the Yak

### LINKED ISSUE #96: Revert OS check when corefx FileVersionInfo bug is fixed
We can revert 129a84130b154df895d5336972f517a7edfc6c2f when dotnet/corefx#11163 is fixed

### PR DESCRIPTION
**Pull request by dsplaisted**
So there I was, looking at #96 and thinking it would be easy to fix.

But when I tried to do so, the test [failed](https://github.com/dotnet/sdk/pull/1202) on non-Windows.

It turned out that this was because our tests still target .NET Core 1.0, and thus weren't getting a version of FileVersionInfo with the fix.

So I retargeted the tests to .NET Core 2.0.

But then they failed at runtime because of the way we were copying all the dependencies into the Tests output directory.

So I ended up doing a whole bunch of clean up on how our tests work:

- They are now mostly normal .NET Core test projects, which work with the test explorer in Visual Studio
- The tests also now use an `OutputType` of `Exe`, and have a Main method which forwards to the xUnit console runner.  This means you can also run the tests with `dotnet run`, and it should be easier to attach a debugger since when run this way it doesn't spawn a separate process to run the tests.
- As part of this, we are now using official xUnit packages instead of xunit.console.netcore

Fixes #96

### CODE REVIEW

**Comment by nguerrera** [comment]
💯

**Comment by nguerrera** [comment]
You've made Jenkins sad, but I'm happy.

**Comment by nguerrera** [comment]
I'll unshelve my build and pack cleanup after this. With the template and copy local shenanigans out of the way, it should be unblocked. :)

**Comment by dsplaisted** [comment]
@nguerrera @livarcocc Should this go into master or into release/2.0.0?

**Comment by nguerrera** [comment]
Master

**Review by nguerrera — APPROVED** [review]

**Comment by nguerrera** [comment]
@dotnet-bot test this please

**Comment by dsplaisted** [comment]
@dotnet-bot test Ubuntu16.04 Debug

**Comment by dsplaisted** [comment]
@dotnet-bot test Windows_NT Release

**Comment by dsplaisted** [comment]
@dotnet-bot test Windows_NT Release

**Comment by dsplaisted** [comment]
@dotnet-bot test Windows_NT_FullFramework Debug

**Comment by dsplaisted** [comment]
@dotnet-bot test Windows_NT_FullFramework Release

File: test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAnAppWithLibrary.cs [hunk 1]
```diff
@@ -128,7 +128,9 @@ public void It_generates_satellite_assemblies()
                 {"", "Welcome to .Net!"},
                 {"da", "Velkommen til .Net!"},
                 {"de", "Willkommen in .Net!"},
-                {"fr", "Bienvenue à .Net!"}
+                //  Only check partial French output due to Console encoding issue: https://github.com/dotnet/cli/issues/6597
+                {"fr", "Bienvenue"}
+                //{"fr", "Bienvenue à .Net!"}
```
Comment (nguerrera) [inline test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAnAppWithLibrary.cs]: We should register the code pages provider instead if we run in .NET Core 1.x. I've closed that issue.

Comment (dsplaisted) [inline test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAnAppWithLibrary.cs]: Any idea why the behavior changed with this PR?  The test app is running on 1.x, but my PR doesn't change that.

Comment (nguerrera) [inline test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAnAppWithLibrary.cs]: The old test runner probably registered it for us.

Comment (dsplaisted) [inline test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAnAppWithLibrary.cs]: Makes sense.  Any thoughts on whether we should register the code pages provider in the test app or just retarget it to .NET Core 2.0?

Comment (nguerrera) [inline test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAnAppWithLibrary.cs]: I'm not sure.

**Comment by dsplaisted** [comment]
@dotnet-bot test this please

**Comment by mmitche** [comment]
@dotnet-bot test this please

**Pull request merged**